lib/ReactGridLayout.jsx

```diff
@@ -429,31 +461,17 @@ export default class ReactGridLayout extends React.Component<Props, State> {
           if (Number.isFinite(leastY)) l.h = leastY - l.y;
         }
       }
+      l.w = w;
+      l.h = h;
 
       return l;
     });
 
     // Shouldn't ever happen, but typechecking makes it necessary
     if (!l) return;
 
-    let finalLayout;
-    if (["sw", "w", "nw", "n", "ne"].indexOf(handle) !== -1) {
-      let x = l.x;
-      let y = l.y;
-      if (["sw", "nw", "w"].indexOf(handle) !== -1) {
-        x = l.x + (l.w - w);
-        w = l.x !== x && x < 0 ? l.w : w;
-        x = x < 0 ? 0 : x;
-      }
-
-      if (["ne", "n", "nw"].indexOf(handle) !== -1) {
-        y = l.y + (l.h - h);
-        h = l.y !== y && y < 0 ? l.h : h;
-        y = y < 0 ? 0 : y;
-      }
-
-      l.w = w;
-      l.h = h;
+    finalLayout = newLayout;
+    if (shouldMoveItem) {
```

Comment:
Similarily, we can't move the item within `withLayoutItem` which is why this is done outside of that block since we need the resulting `Layout` to perform the operation.